python_modules/dagster/dagster/_core/definitions/multi_asset_sensor_definition.py

```diff
@@ -125,7 +125,7 @@ class MultiAssetSensorContextCursor:
     # Must call MultiAssetSensorEvaluationContext._update_cursor_after_evaluation at end of tick
     # to serialize the cursor.
     def __init__(self, cursor: Optional[str], context: "MultiAssetSensorEvaluationContext"):
-        loaded_cursor = json.loads(cursor) if cursor else {}
+        loaded_cursor = json.loads(cursor) if cursor and hasattr(cursor, "items") else {}
```

Comment:
At this point in `hasattr(cursor, "items")` cursor is still a string, so I don't believe that it has an `items` attribute. Should this instead be something like:
```python
loaded_cursor = json.loads(cursor) if cursor else {}
loaded_cursor = loaded_cursor if isinstance(loaded_cursor, dict) else {}
```

Reply:
> At this point in hasattr(cursor, "items") cursor is still a string, so I don't believe that it has an items attribute.

That means my change accidentally "fixed" my problem because Dagster actually never loaded any cursor for that sensor is that right ? I thought I waited enough to eventually see the sensor kick at every tick, if the cursor handling was wrong. It this not what should happen if the cursor was never loaded ?

Reply:
yeah -- I believe the original `hasattr(cursor, "items")` always returns False, which means that an empty dict is always loaded as the cursor. This is fine for the first tick, but the cursor thereafter from the previous tick is never used.